intel-sgx/pcs/src/tcb_info.rs

```diff
@@ -114,11 +114,99 @@ impl<'de> Deserialize<'de> for Fmspc {
 #[derive(Serialize, Deserialize, Clone, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct TcbLevel {
-    pub tcb: TcbComponents,
-    pub tcb_date: String,
-    pub tcb_status: TcbStatus,
+    tcb: TcbComponents,
+    tcb_date: String,
+    tcb_status: TcbStatus,
     #[serde(default, rename = "advisoryIDs", skip_serializing_if = "Vec::is_empty")]
-    pub advisory_ids: Vec<String>,
+    advisory_ids: Vec<AdvisoryID>,
+}
+
+impl TcbLevel {
+    pub fn components(&self) -> &TcbComponents {
+        &self.tcb
+    }
+
+    pub fn tcb_status(&self) -> TcbStatus {
+        self.tcb_status
+    }
+
+    pub fn advisory_ids(&self) -> &Vec<AdvisoryID> {
+        &self.advisory_ids
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum AdvisoryID {
+    /// Security Advisory ID - "INTEL-SA-XXXXX" (where XXXXX is a placeholder for a 5-digit
+    /// number) - representing Security Advisories that can be searched on Intel® Product
+    /// Security Center Advisories page
+    /// (https://www.intel.com/content/www/us/en/security-center/default.html)
+    Security(u32),
+    /// Document Advisory ID - "INTEL-DOC-XXXXX" (where XXXXX is a placeholder for a 5-digit
+    /// number) - representing articles containing additional information about the attested
+    /// platform. The articles can be found under the following URL:
+    /// https://api.trustedservices.intel.com/documents/{docID}
+    Documentation(u32)
+}
+
+impl ToString for AdvisoryID {
+    fn to_string(&self) -> String {
+        match self {
+            AdvisoryID::Security(v) => format!("INTEL-SA-{:05}", v),
+            AdvisoryID::Documentation(v) => format!("INTEL-DOC-{:05}", v),
+        }
+    }
+}
+
+impl TryFrom<&str> for AdvisoryID {
+    type Error = &'static str;
+
+    fn try_from(s: &str) -> Result<AdvisoryID, &'static str> {
+        fn tokenize(s: &str) -> Result<(String, String, u32), &'static str> {
+            let mut chunks = s.trim().split('-');
+
+            let intel = chunks.next().ok_or("Couldn't parse advisory ID number")?;
```

Comment:
It might be helpful if the logs could distinguish which chunk was not available by changing the error message here i.e. `Couldn't parse <intel/type/value> of the advisory ID number` in lines 168-171.